### DISCUSSION TITLE
Configurable Spend cap / Budget / Usage limits

### DISCUSSION THREAD

**Discussion by Strajk** [supabase/supabase#14356] [Feature Requests]
My use case:
I have a Pro plan, I like the safety of Spend cap but I would like to pay for more storage (e.g. 50GB instead of 8GB included in the plan).

Ideally, I should be able to keep the Spend cap (= disable ops in Supabase account on overage), but pay 42*$0.125 extra for storage. 

Less ideally, but still great, I should be able to set the Spend cap to e.g. $100/month.

**Comment by pierroo** [comment]
I was going to create a new discussion for that but then found this thread, EXACTLY what I was wondering about.

The binary choice HARD CAP / NO CAP AT ALL is super risky in both scenarios.

Anyway to adjust manually the cursor with a specific amount cap beyond the default one?
@GaryAustin1 if you happen to pass by... :)

**Comment by kevcodez** [comment]
More sophisticated cost control is planned for 2025 - we also want to move away from the hard cap / no cap at all model, towards something more budget based with soft/hard cap. No promises on the timeline yet, but definitely something we will tackle ;)

**Reply by activenode** [reply]
awesome!

**Reply by felipenmoura** [reply]
> More sophisticated cost control is planned for 2025 - we also want to move away from the hard cap / no cap at all model, towards something more budget based with soft/hard cap. No promises on the timeline yet, but definitely something we will tackle ;)

I left a comment in this issue but I felt compelled to mention it here too, in case it might help this out :)

This is a feature that every platform includes and...let's get real here...I was **embarrassed** when a client asked me "where was the settings for that" and I hade to, with a very awkward smile, say it was not possible :(

I, developer, really enjoy supabase. But I, businessman, sometimes face some hard times convincing clients because of that!
Clients will simply go for an _on premise_ alternative (or another SaaS), obviously

C'mon ... help me help you :)

**Comment by AdminPatentPlus** [comment]
How is it possible that this does not already exist? Obviously I want to set a max possible spend per month in USD as a user.

**Comment by iveshenry18** [comment]
Hi, just want to add an emphatic +1 on this. Surely many paying customers would value being able to set a cap that allows us to pay a capped amount above the $25/month without exposing ourselves to the risk of uncapped spending.

I'm building a social app with image-sharing, so I frequently am near or slightly above the Storage Image Transformations quota for the Pro tier, while being far below all other quotas.

It's hugely unfortunate that I have to expose myself and my company to the risk of an unexpectedly large uncapped bill just to be able to pay the extra $5 for more Storage Image Transformations.

 I understand that it's hard to prioritize a feature like this (it doesn't _directly_ drive customers or revenue, and it's arguably _disadvantageous_ to certain modes of revenue), but count this as a strong request for the feature!

**Reply by lukajose** [reply]
Any updates on this issue? I have the same problem

**Comment by albertlinao** [comment]
Any updates on this from the Supabase team? This issue's been up for years and is still being requested by different users regularly.

**Reply by kevcodez** [reply]
No news on this unfortunately - still planned but currently deprioritized with competing priorities. We have a few new folks on the billing team now, so we'll hopefully get to that sooner than later.

**Reply by johnleonard1** [reply]
yeah this is crazy that there's not a "notify me when 80% limit is reached". every other platform does this :/

**Comment by realfishsam** [comment]
+1

**Comment by felipenmoura** [comment]
Ok ... so this is now 2026.

Any update on that please?
Two very, **_VERY_** important and **NECESSARY** features to be added in the pro plan and above:

- how much I am willing to pay on a given project
- tell me when 80% of  that is reached

As a SaaS, specially nowadays, I do believe this is very much a fundamental feature.

If the risk is related to people fine-tuning their costs instead of going to the Team tier, you could add a limit, like allowing the customization up to a given amount, for example.

**Comment by RomainPln** [comment]
Any update on this? 
This is a fundamental requirement for our clients. "**Infinite risk**", or call it **bankruptcy**, is a _deal breaker_ to them.
Please prioritise this.

**Comment by marco242424** [comment]
Still waiting for this...
